From alan@linux.intel.com  Wed Jan 27 15:38:21 2010
From: Alan Cox <alan@linux.intel.com>
Date: Mon, 18 Jan 2010 15:32:44 +0000
Subject: Staging: et131x: kill of FBR_WORD2
To: greg@kroah.com
Message-ID: <20100118153244.6884.12955.stgit@localhost.localdomain>


This is only used meaningfully as a definition, we never mask and fetch the
bits apart

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/et131x/et1310_rx.c |    8 ++++----
 drivers/staging/et131x/et1310_rx.h |   15 +--------------
 2 files changed, 5 insertions(+), 18 deletions(-)

--- a/drivers/staging/et131x/et1310_rx.c
+++ b/drivers/staging/et131x/et1310_rx.c
@@ -640,7 +640,7 @@ void ConfigRxDmaRegs(struct et131x_adapt
 	for (entry = 0; entry < rx_local->Fbr1NumEntries; entry++) {
 		fbr_entry->addr_hi = rx_local->Fbr[1]->PAHigh[entry];
 		fbr_entry->addr_lo = rx_local->Fbr[1]->PALow[entry];
-		fbr_entry->word2.bits.bi = entry;
+		fbr_entry->word2 = entry;
 		fbr_entry++;
 	}
 
@@ -665,7 +665,7 @@ void ConfigRxDmaRegs(struct et131x_adapt
 	for (entry = 0; entry < rx_local->Fbr0NumEntries; entry++) {
 		fbr_entry->addr_hi = rx_local->Fbr[0]->PAHigh[entry];
 		fbr_entry->addr_lo = rx_local->Fbr[0]->PALow[entry];
-		fbr_entry->word2.bits.bi = entry;
+		fbr_entry->word2 = entry;
 		fbr_entry++;
 	}
 
@@ -1126,7 +1126,7 @@ void nic_return_rfd(struct et131x_adapte
 			 */
 			next->addr_hi = rx_local->Fbr[1]->PAHigh[bi];
 			next->addr_lo = rx_local->Fbr[1]->PALow[bi];
-			next->word2.value = bi;
+			next->word2 = bi;
 
 			writel(bump_fbr(&rx_local->local_Fbr1_full,
 				rx_local->Fbr1NumEntries - 1),
@@ -1144,7 +1144,7 @@ void nic_return_rfd(struct et131x_adapte
 			 */
 			next->addr_hi = rx_local->Fbr[0]->PAHigh[bi];
 			next->addr_lo = rx_local->Fbr[0]->PALow[bi];
-			next->word2.value = bi;
+			next->word2 = bi;
 
 			writel(bump_fbr(&rx_local->local_Fbr0_full,
 					rx_local->Fbr0NumEntries - 1),
--- a/drivers/staging/et131x/et1310_rx.h
+++ b/drivers/staging/et131x/et1310_rx.h
@@ -91,23 +91,10 @@
 #define ALCATEL_BROADCAST_PKT	0x02000000
 
 /* typedefs for Free Buffer Descriptors */
-typedef union _FBR_WORD2_t {
-	u32 value;
-	struct {
-#ifdef _BIT_FIELDS_HTOL
-		u32 reserved:22;	/* bits 10-31 */
-		u32 bi:10;		/* bits 0-9(Buffer Index) */
-#else
-		u32 bi:10;		/* bits 0-9(Buffer Index) */
-		u32 reserved:22;	/* bit 10-31 */
-#endif
-	} bits;
-} FBR_WORD2_t, *PFBR_WORD2_t;
-
 typedef struct _FBR_DESC_t {
 	u32 addr_lo;
 	u32 addr_hi;
-	FBR_WORD2_t word2;
+	u32 word2;		/* Bits 10-31 reserved, 0-9 descriptor */
 } FBR_DESC_t, *PFBR_DESC_t;
 
 /* Typedefs for Packet Status Ring Descriptors */
